src/modules/marketplaces/marketplaces-events-indexing.service.ts

```diff
@@ -85,7 +81,10 @@ export class MarketplaceEventsIndexingService {
         throw new Error(`beforeTimestamp can't be less than afterTimestamp`);
       }
 
-      const [newestTimestamp] = await this.getEventsAndSaveToDb(input);
+      let newestTimestamp: number;
+
```

Comment:
why here you rewrite the newestTimestamp? Are there moments when await this.getTxAndSaveToDb(input); returns null for timestamp?

Reply:
Fixed